### PR TITLE
Relay main

### PR DESCRIPTION
**Pull request by cody-littley**
## Why are these changes needed?

This PR adds a main function and GRPC server for the relay.

Not yet ready for review, needs https://github.com/Layr-Labs/eigenda/pull/869 to merge first.

## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: go.mod
Comment (cody-littley) [inline go.mod:1]: These changes were the result of installing viper via `go get github.com/spf13/viper`

https://github.com/spf13/viper

File: relay/config.go [hunk 1]
```diff
@@ -1,14 +1,61 @@
...
+const (
+	// relayEnvPrefix is the prefix for all environment variables used by the relay.
+	relayEnvPrefix = "RELAY"
+)
 
 // Config is the configuration for the relay Server.
+//
+// Environment variables are mapped into this struct by taking the name of the field in this struct,
+// converting to upper case, and prepending "RELAY_". For example, "BlobCacheSize" can be set using the
+// environment variable "RELAY_BLOBCACHESIZE".
```
Comment (jianoaix) [inline relay/config.go:22]: Is it a constraint of viper? This is slightly different format than what we currently use, which is underscore connected, e.g. `MAX_BLOB_SIZE`.

File: relay/config.go [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+	AWS aws.ClientConfig
+
 	// RelayIDs contains the IDs of the relays that this server is willing to serve data for. If empty, the server will
 	// serve data for any shard it can.
 	RelayIDs []core.RelayKey
 
+	// GRPCPort is the port that the relay server listens on. Default is 50051. // TODO what is a good port?
+	GRPCPort int
+
+	// BucketName is the name of the S3 bucket that stores blobs. Default is "relay".
```
Comment (jianoaix) [inline relay/config.go:49]: Are chunks and blobs in different buckets? If so, might be better to make the names more specific

**Review by jianoaix — APPROVED** [review]

**Review by ian-shim — APPROVED** [review]

File: relay/config.go (hunk 2 above)
Comment (cody-littley) [reply relay/config.go:49]: No, they are in the same bucket. We discussed this as a group when you were on PTO, and came to the conclusion that it was simpler to use a single bucket. We use [this code](https://github.com/Layr-Labs/eigenda/blob/master/common/aws/s3/scoped_keys.go) to ensure we don't have collisions between different data types.

File: relay/config.go (hunk 1 above)
Comment (cody-littley) [reply relay/config.go:22]: There is a way, but lets leave that to a follow up PR (as discussed offline).

**Pull request merged**